.github/workflows/test_suite.yml

```diff
@@ -24,12 +24,25 @@ jobs:
     # - name: Fix Conda permissions on macOS
     #   run: sudo chown -R $UID $CONDA
     #   if: runner.os == 'macOS'
+
     - name: Install IC
       run: |
         source $CONDA/etc/profile.d/conda.sh
         source manage.sh work_in_python_version_no_tests ${{ matrix.python-version }}
+
+
     - name: Run tests
       run: |
+        set -o pipefail
         source $CONDA/etc/profile.d/conda.sh
         source manage.sh work_in_python_version_no_tests ${{ matrix.python-version }}
-        PYTEST_ADDOPTS=--color=yes HYPOTHESIS_PROFILE=travis-ci bash manage.sh run_tests_par
+        PYTEST_ADDOPTS=--color=yes HYPOTHESIS_PROFILE=travis-ci bash manage.sh run_tests_par | tee pytest_output.txt
+
+    - name: Warning summary
+      run: |
+        ./scripts/check_warnings pytest_output.txt | tee warnings
+        if [ -s warnings ]; then
+            echo "::warning ::There are warnings raised in the test suite"
+        else
+            echo "No warnings were raised."
+        fi
```

Comment:
It feels a bit silly and possible convoluted to ask for a test of a warning script for tests. But is there any way we could test that this output is provided if an error does occur in pytest? Possibly by writing a false warning to a file and running `check_warnings` on it to ensure you get the appropriate output from the GHA?

Reply:
Hmm, I agree it would be nice, but I'm not sure how to do it without making it silly. I tested it in my fork before opening the PR. See for instance [this](https://github.com/gonzaponte/IC/actions/runs/11820831339) workflow run. Images attached for reference.
![image](https://github.com/user-attachments/assets/10b740cb-efa4-48a9-b6db-a76334b59bac)
![image](https://github.com/user-attachments/assets/be069a94-bb8e-4d83-bc54-02f7cb0b8680)

Reply:
That is quite a large output for a single test case. My guess would be that in the GHA you would run `pytest` in a particular directory with a single (possibly a dummy) test that is guaranteed to produce a warning, and check the corresponding behaviour from `check_warnings` occurs. This does feel a bit overbearing though, so if you think it's too much I'm happy to approve the PR :)

Reply:
Just to clarify, this is the output for the full test suite. I've added a test in the workflow. This is to avoid having unnecessary files in the repo, so they are created on the fly and they only exist in the GHA runner.